swarm/pss/pss.go

```diff
@@ -360,10 +627,14 @@ func (self *Pss) Forward(msg *PssMsg) error {
 // Links a remote peer and Topic to a dedicated p2p.MsgReadWriter in the pss peerpool, and runs the specificed protocol using these resources.
 //
 // The effect is that now we have a "virtual" protocol running on an artificial p2p.Peer, which can be looked up and piped to through Pss using swarm overlay address and topic
-func (self *Pss) AddPeer(p *p2p.Peer, addr pot.Address, run func(*p2p.Peer, p2p.MsgReadWriter) error, topic Topic, rw p2p.MsgReadWriter) error {
+//
+// The peer's encryption keys must be added separately.
+func (self *Pss) AddPeer(p *p2p.Peer, addr pot.Address, run func(*p2p.Peer, p2p.MsgReadWriter) error, topic whisper.TopicType, rw p2p.MsgReadWriter) error {
 	self.lock.Lock()
 	defer self.lock.Unlock()
-	self.addPeerTopic(addr, topic, rw)
+	self.preparePeerTopic(addr, topic)
```

Comment:
well otherwise the next line you need to assign it to a variable by retrieving from the map. 
feels more natural to return it by the findOrCreatePeer call which it should be called